static/app/views/starfish/components/samplesTable/useSlowMedianFastSamplesQuery.tsx

```diff
@@ -0,0 +1,163 @@
+    referrer: 'starfish-transaction-summary-sample-events',
+    location,
+    orgSlug: organization.slug,
+  });
+
+  const slowestSamplesEventView = eventView
+    .clone()
+    .withColumns(commonColumns)
+    .withSorts([
+      {
+        field: 'transaction.duration',
+        kind: 'desc',
+      },
+    ]);
+
+  slowestSamplesEventView.additionalConditions = new MutableSearch(
+    `transaction.duration:>${
+      aggregatesData?.data?.[0]?.['p95(transaction.duration)'] ?? 0
+    }`
+  );
+
+  const {isLoading: isLoadingSlowest, data: slowestSamplesData} = useGenericDiscoverQuery<
+    any,
+    DiscoverQueryProps
+  >({
+    route: 'events',
+    eventView: slowestSamplesEventView,
+    referrer: 'starfish-transaction-summary-sample-events',
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...slowestSamplesEventView.getEventsAPIPayload(location),
+    }),
+    limit: LIMIT_PER_POPULATION,
+  });
+
+  const medianSamplesEventView = eventView
+    .clone()
+    .withColumns(commonColumns)
+    .withSorts([
+      {
+        field: 'transaction.duration',
+        kind: 'desc',
+      },
+    ]);
+
+  medianSamplesEventView.additionalConditions = new MutableSearch(
+    `transaction.duration:<=${
+      aggregatesData?.data?.[0]?.['p50(transaction.duration)'] ?? 0
+    }`
+  );
+
+  const {isLoading: isLoadingMedian, data: medianSamplesData} = useGenericDiscoverQuery<
+    any,
+    DiscoverQueryProps
+  >({
+    route: 'events',
+    eventView: medianSamplesEventView,
+    referrer: 'starfish-transaction-summary-sample-events',
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...medianSamplesEventView.getEventsAPIPayload(location),
+    }),
+    limit: LIMIT_PER_POPULATION,
+  });
+
+  const fastestSamplesEventView = eventView
+    .clone()
+    .withColumns(commonColumns)
+    .withSorts([
+      {
+        field: 'transaction.duration',
+        kind: 'asc',
+      },
+    ]);
+
+  fastestSamplesEventView.additionalConditions = new MutableSearch(
+    `transaction.duration:<=${
+      aggregatesData?.data?.[0]?.['p50(transaction.duration)'] ?? 0
+    }`
+  );
+
+  const {isLoading: isLoadingFastest, data: fastestSamplesData} = useGenericDiscoverQuery<
+    any,
+    DiscoverQueryProps
+  >({
+    route: 'events',
+    eventView: fastestSamplesEventView,
+    referrer: 'starfish-transaction-summary-sample-events',
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...fastestSamplesEventView.getEventsAPIPayload(location),
+    }),
+    limit: LIMIT_PER_POPULATION,
+  });
+
+  if (isLoadingAgg || isLoadingSlowest || isLoadingMedian || isLoadingFastest) {
+    return {isLoading: true, data: []};
+  }
+
+  const combinedData = [
+    ...slowestSamplesData.data,
+    ...medianSamplesData.data,
+    ...fastestSamplesData.data,
+  ];
+
+  return {isLoading: false, data: combinedData, aggregatesData: aggregatesData.data[0]};
```

Comment:
Just typical code crimes lol, we do this in a lot of other places in our app. I don't want to invest too much time into those details since it's uncertain whether this code is going stay, we can certainly clean it up later if it does